src/GameLogic/Entity.ts

```diff
@@ -1,9 +1,10 @@
 import {EntityType} from './Enums';
+import {Level} from './Level';
 
 export interface Entity {
 	readonly type: EntityType;
 	x: number;
 	y: number;
 
-	update(): void;
+	update(level: Level): void;
```

Comment:
So we're passing the entire level into every entity? Is this a good idea? I would have thought the level contained the entities. Perhaps I misunderstand.